daml-script/dump/src/main/scala/com/daml/script/dump/Dump.scala

```diff
@@ -36,26 +33,31 @@ object Dump {
         .render(80)
         .getBytes(StandardCharsets.UTF_8),
     )
-    val dars: Seq[Dar[(PackageId, ByteString, Ast.Package)]] =
-      pkgRefs.view.collect(Function.unlift(Dependencies.toDar(_, pkgs))).toSeq
+    val packages: Seq[String] =
+      pkgRefs.view.collect(Function.unlift(Dependencies.toPackages(_, pkgs))).toSeq
+    val dalfs: Seq[(PackageId, ByteString, Ast.Package)] =
+      pkgRefs.view.flatMap(Dependencies.toDalfs(_, pkgs)).toSeq
```

Comment:
I don’t quite understand this.
Before we were looking at the direct references producing one DAR for each of them because the main dalf of each of those gets exposed.
Now we produce the direct references in `packages`. That’s all good. But then we chase dependencies for each of those to produce `dalfs` which can contain duplicate packages afaik. Don’t we already know that we need all the dalfs in `pkgs`? Can we just map over those filtering out a few and kill the `toDalfs` stuff?